api/internal/controllers/analysis/analysis.go

```diff
@@ -147,11 +158,96 @@ func (c *Controller) hasDuplicatedHash(
 }
 
 func (c *Controller) publishInBroker(analysisID uuid.UUID) error {
-	response, err := c.GetAnalysis(analysisID)
+	res, err := c.GetAnalysis(analysisID)
 	if err != nil {
 		return err
 	}
 
 	return c.broker.Publish("", exchange.NewAnalysis,
-		exchange.Fanout, response.ToBytes())
+		exchange.Fanout, res.ToBytes())
+}
+
+// TODO:REMOVE ALL BELOW AFTER v2.10.0
+// treatCompatibility checks if the field Analysis.AnalysisVulnerabilities[i].DeprecatedHashes exists
+// and if so, find them on database and updates it with the correct field Analysis.AnalysisVulnerabilities[i].VulnHash.
+// this is only a temporary fix to maintain compatibility between versions,
+// it will be deleted when v2.10.0 is released
+func (c *Controller) treatCompatibility(analysisEntity *analysis.Analysis) error {
+	if !c.existsDepecratedHashesSlice(analysisEntity.AnalysisVulnerabilities) {
+		return nil
+	}
+	hashSlice := make([]string, 0)
```

Comment:
I think that `deprecatedHashes` it's a better name than `hashSlice`.

Since we have both valid and deprecated hashes it's best to be clear about which one we're manipulating.